### ISSUE TITLE
CA1508 seems to assume that `is var` will return false for null values.

### ISSUE THREAD

**Issue by brian-reichle** [dotnet/roslyn-analyzers#5154]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/5.0.3)

**Version**: 5.0.3 (Latest)

### Describe the bug

In the following code, CA1508 will complain that `bar != null` is always true, but this is incorrect as `is var` does not check for null. If you enable NRT and hover over `bar`, the ide will correctly report that `'bar' may be null here.`

```csharp
public static class Foo
{
	static object? Bar() => null;

	public static void Baz()
	{
		if (Bar() is var bar &&
			bar != null) // <-- reports error here.
		{
			bar.ToString(); // <-- without the null check, NRT will complain about dereferencing a possibly null reference here.
		}
	}
}
```

If I remove the null check, then NRT will complain about dereferencing a possibly null reference on `bar.ToString()`.

### Steps To Reproduce

Create a new project with NetAnalyzers v5.0.3 and add the above code.

### Expected behavior

Should not report the CA1508 diagnostic on the `bar != null` expression.

### Actual behavior

Reports a CA1508 diagnostic on the `bar != null` expression.
